### PR TITLE
update Costmap API documentation

### PR DESCRIPTION
**Pull request by JacksonK9**
update Costmap API documentation as the Pull Request of navigation2

* ros-planning/navigation2#3169

### CODE REVIEW

File: commander_api/index.rst [hunk 1]
```diff
@@ -147,6 +147,23 @@ This is a Python3 API for costmap 2d messages from the stack. It provides the ba
 +---------------------------------------+----------------------------------------------------------------------------+
 | getCostmapTimestamp()                 | Get costmap timestamp.                                                     |
 +---------------------------------------+----------------------------------------------------------------------------+
+| getCostXY(mx, my)                     | Get the cost (``int``) of a cell in the costmap using mx(``int``),         |
+|                                       | my(``int``) of Map Coordinate.                                             |
++---------------------------------------+----------------------------------------------------------------------------+
+| getCostIdx(index)                     | Get the cost (``int``) of a cell in the costmap using index(``int``)       |
++---------------------------------------+----------------------------------------------------------------------------+
+| setCost(mx, my, cost)                 | Set the cost (``np.int8``) of a cell in the costmap using mx(``int``),     |
```
Comment (SteveMacenski) [inline commander_api/index.rst:155]: Why is this one np.int8 but others are just straight ints?

File: commander_api/index.rst [hunk 2]
```diff
@@ -147,6 +147,23 @@ This is a Python3 API for costmap 2d messages from the stack. It provides the ba
 +---------------------------------------+----------------------------------------------------------------------------+
 | getCostmapTimestamp()                 | Get costmap timestamp.                                                     |
 +---------------------------------------+----------------------------------------------------------------------------+
+| getCostXY(mx, my)                     | Get the cost (``int``) of a cell in the costmap using mx(``int``),         |
```
Comment (SteveMacenski) [inline commander_api/index.rst:150]: Make the units similar to above, where they're in square brackets, ex

Get the cost [int] of a cell in the costmap using mx [int].

Comment (SteveMacenski) [reply commander_api/index.rst:150]: And make sure to have a space between the "thing" (mx, my, etc) and the units. Some are butted against each other

**Review by SteveMacenski — CHANGES_REQUESTED** [review]

File: commander_api/index.rst (hunk 1 above)
Comment (JacksonK9) [reply commander_api/index.rst:155]: The type of costmap is np.array of np.int8.
So I pointed that detail, But we can use just straight int.
Do you think I should change to Int? or Just leave it?

Comment (SteveMacenski) [reply commander_api/index.rst:155]: uint seems right, since negative values are invalid

Comment (JacksonK9) [reply commander_api/index.rst:155]: Okay I will change all the int type to np.uint8 and notify again.
Thank you for your comment 👍

**Comment by JacksonK9** [comment]
I think document show the type with (``np.uint8``) and unit with [m].

So I add the type and unit to the document of commander api.

Thank you.

File: commander_api/index.rst [hunk 3]
```diff
@@ -147,6 +147,23 @@ This is a Python3 API for costmap 2d messages from the stack. It provides the ba
 +---------------------------------------+----------------------------------------------------------------------------+
 | getCostmapTimestamp()                 | Get costmap timestamp.                                                     |
 +---------------------------------------+----------------------------------------------------------------------------+
+| getCostXY(mx, my)                     | Get the cost (``np.uint8``) of a cell in the costmap using mx (``int``)    |
```
Comment (SteveMacenski) [inline commander_api/index.rst:150]: again, units in square brackets.

**Review by SteveMacenski — CHANGES_REQUESTED** [review]

File: commander_api/index.rst [hunk 4]
```diff
@@ -147,6 +147,23 @@ This is a Python3 API for costmap 2d messages from the stack. It provides the ba
...
+|                                       | , my (``int``) of Map Coordinate.                                          |
++---------------------------------------+----------------------------------------------------------------------------+
+| getCostIdx(index)                     | Get the cost (``np.uint8``) of a cell in the costmap using index (``int``) |
```
Comment (SteveMacenski) [inline commander_api/index.rst:153]: Again, why are we mixing uints and ints?

**Review by SteveMacenski — CHANGES_REQUESTED** [review]

File: commander_api/index.rst (hunk 3 above)
Comment (SteveMacenski) [reply commander_api/index.rst:150]: Also, the actual API implementation uses `np.int8` not unsigned ints

Comment (JacksonK9) [reply commander_api/index.rst:150]: As I check these two lines.

setInitialPose(initial_pose)          | Sets the initial pose (``PoseStamped``) of the robot to localization.  

getResolution()                       | Get map resolution [m/cell].      

So I think the "variable type" is using parentheses and the "unit" is using square bracket.

As a result, I change like the below line.

mapToWorld(mx, my)                    | Get the wx (``float``) [m], wy (``float``) [m] of world coordinate XY using

Should I change to "Get the wx [``float``] [m], wy [[``float``] [m] of world coordinate XY using" ?

Can you tell me which one is better?

Thank you for your comment.

**Review by SteveMacenski — APPROVED** [review]

**Pull request merged**